scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/component/parameter/validator/ValidationExpressionParameterValidator.scala

```diff
@@ -22,6 +22,17 @@ case class ValidationExpressionParameterValidator(
       implicit nodeId: NodeId
   ): Validated[PartSubGraphCompilationError, Unit] = {
     value match {
+      case None if !expression.expression.isBlank =>
```

Comment:
Currently, our validators are composable. Thanks to that we avoid duplicated errors from different validators. To achieve that, each validator has to ignore the scope of validation that it is not responsible for.  Because of that we have this `isBlank` checking. @bartektartanus prepared validator that check the same scope as you check here - `CompileTimeEvaluableValueValidator`, I think that it was designed to be composed with your validator. But TBH I'm not sure if it used anywhere else than in tests. @bartektartanus can you help in this area? If it isn't used, we could just merge them. If it is used, we should think a while how to connect them. Because this validator has no sense without validation done in `CompileTimeEvaluableValueValidator`

Reply:
i've nested `CompileTimeEvaluableValueValidator` inside `ValidationExpressionParameterValidator`, my logic being that `CompileTimeEvaluableValueValidator` makes sense standing on it's own, while `ValidationExpressionParameterValidator` always requires `CompileTimeEvaluableValueValidator`

Reply:
:thinking: I guess this can still duplicate errors if both are present

Reply:
You have some test failures

Reply:
yep i renamed the error and didn't rename the expected error name, classic